src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -786,7 +790,7 @@ class ColumnConstraintSegment(BaseSegment):
     This matches the definition in https://www.postgresql.org/docs/13/sql-altertable.html
     """
 
-    type = "column_constraint"
+    type = "column_constraint_segment"
```

Comment:
This is called `column_coinstraint` in ANSI. Should we leave as is, or change both?

Reply:
In my opinion we should go for column_constraint and table_constraint respectively. We aren't necessarily consistent with type naming, but those would follow closest to the general theme

Reply:
I agree yours are better names. I don’t agree with being inconsistent between dialects for the same thing.

So, as I say, happy either reverting to the slightly worse names. Or updating the ANSI dialect to the new names as part of this PR.

Reply:
I agree with @tunetheweb: Names (and parse trees) should be consistent across dialects whenever practical, because the lint rules examine that structure when looking for issues. We don't want the rules to have to worry about unnecessary differences between dialects.

Reply:
Yeah, just pushed a commit aligning them for TableConstraintSegment and ColumnConstraintSegment across the dialects